### PR TITLE
Finish run file read in

### PR DESCRIPTION
**Pull request by znicholls**
# New Feature

This will ensure that we can read every file in the `pymagicc/MAGICC6/run` directory with our `MAGICCInput` class into a common format.

Please confirm that this pull request has done the following:

- [x] Tests added
- [x] Documentation added
- [x] Example added (either to an existing notebook or as a new notebook)
- [x] Description in CHANGELOG.md added
- [ ]  Branch rebased

## Adding to CHANGELOG.md

Please add a single line in the changelog notes similar to the following:

```
- (#XX)[http://link-to-pr.com] Added feature which does something
```

### CODE REVIEW

**Comment by znicholls** [comment]
@rgieseke crash course on how the docs work?

**Comment by znicholls** [comment]
this will also now close #112

**Comment by znicholls** [comment]
@rgieseke going to need your help with this notebook CI stuff...

**Comment by rgieseke** [comment]
Re Notebooks ... I'll have a look, seems only 3.6 failed ...

**Comment by rgieseke** [comment]
> Re Notebooks ... I'll have a look, seems only 3.6 failed ..

Ok, notebooks are only tested on 3.6 :-]

**Comment by znicholls** [comment]
> Ok, notebooks are only tested on 3.6 :-]

lol all good

**Comment by znicholls** [comment]
currently blocked by #114

**Comment by rgieseke** [comment]
Nope, not yet ... still trying to figure out what's going on ...

**Comment by rgieseke** [comment]
Pushed a potential fix, fingers crossed :-)

**Comment by znicholls** [comment]
@rgieseke all yours

File: pymagicc/MAGICC6/run/HISTRCP_HFC245fa_CONC.IN [hunk 1]
```diff
@@ -1,91 +1,91 @@
...
-       1995     0.00000000e+000  
-       1996     0.00000000e+000  
-       1997     0.00000000e+000  
-       1998     0.00000000e+000  
-       1999     0.00000000e+000  
-       2000     0.00000000e+000  
+
+__RCP Historical Concentration File: HFC245fa__
+
+Data: Global average mixing ratio by näme with ümlauts ëh ça
```
[the remark this answers is not in the document]
Comment (znicholls) [reply pymagicc/MAGICC6/run/HISTRCP_HFC245fa_CONC.IN]: good pick up

**Comment by znicholls** [comment]
> There are a few name changes, are these obvious errors? Like FORC to OT or EDGAR_NOX_EMIS_FOSSIL&IND to EDGAR_NOXI_EMIS

FORC to OT is an obvious error (the data in the file was mislabelled at writing time).

EDGAR_NOX_EMIS_FOSSIL&IND to EDGAR_NOXI_EMIS just updates the files to follow MAGICC6 conventions. Dealing with MAGICC5 conventions too is possible but I'd rather not given it's only a few files... Will change if you'd like

**Comment by rgieseke** [comment]
Could you document the fixed categories in the Changelog? That's probably good to have documented.

**Comment by znicholls** [comment]
> Could you document the fixed categories in the Changelog?

Yep, they should probably also go elsewhere, in the docs? In the docstring of MAGICCInput?

**Comment by rgieseke** [comment]
> Yep, they should probably also go elsewhere, in the docs? In the docstring of MAGICCInput?

You mean all of them? For the changed ones I think mentioning them in the changelog is enough.
Might be good to have a central location with more details for all inputs. Not sure if we should put this in Pymagicc docs or this would duplicate info available at magicc.org or in the MAGICC repo.

**Comment by znicholls** [comment]
> Might be good to have a central location with more details for all inputs. Not sure if we should put this in Pymagicc docs or this would duplicate info available at magicc.org or in the MAGICC repo.

It doesn't exist anywhere else at the moment. Given that all of this stuff is related to input/output (which is/should be almost exclusively pymagicc's domain) I think we should put it here. Do you agree?

**Comment by rgieseke** [comment]
>  I think we should put it here. Do you agree?

Yes, this is the only public place under version control so it's the best place!

**Comment by rgieseke** [comment]
Is this still "WIP"? Do i need to approve of a change here?

**Comment by znicholls** [comment]
> Is this still "WIP"? Do i need to approve of a change here?

Yes I still see 'Changes requested' here

**Comment by rgieseke** [comment]
Alright, I see now -- would be good to have the changed/fixed names listed in the Changelog.

**Comment by rgieseke** [comment]
I added a tiny note (8bdb643) ... is this ready to be merged from your side?

**Comment by rgieseke** [comment]
Not sure whether it was my fiddling in #127 that broke Appveyor ...

**Comment by znicholls** [comment]
> Alright, I see now -- would be good to have the changed/fixed names listed in the Changelog.

Done. How can I put something like the below in the docs?

The MAGICC `run` directory contains all of MAGICC's input files. At the moment, we have only tested reading of these files. Hence whilst we can also write these files, we make guarantee that having written them, they will a) run MAGICC or b) be handled in the expected way. 

The files in the `run` directory follow a number of conventions, we summarise these here (please note that this is just a summary, we are still unsure about how strictly many of these conventions have to be followed as we are yet to set up tests for running MAGICC with all of these different files after they've been set using pymagicc).

1. Files ending in `.SCEN` are emissions scenario files. They are used to drive MAGICC6 and can also be used to drive MAGICC7, although it is much less predictable with MAGICC7.
1. Files ending in `.SCEN7` and `.IN` are other input files. They are both the same format, some notes, then the data in space separated columns.
    1. Files ending in `.SCEN7` are scenario input files.
    1. Files ending in `.IN` are input files, typically related to the historical period. The filename indicates what data is in the file e.g. `CONC.IN` indicates that this file contains concentrations. 
1. Files ending in `.CFG` are config files. They are used to set MAGICC's configuration and are always in the format of a FORTRAN90 namelist.
1. Files ending in `.prn` are emissions input files for ozone depleting substances. This file format comprises of a header row which contains (unlabelled) information about the first data row, start year and end year, then some notes rows, then the data in fixed width columns. This file format is used by MAGICC6 but has been deprecated for MAGICC7 in favour of `*EMIS.IN` and `*.SCEN7` files.

**Comment by rgieseke** [comment]
> Done. How can I put something like the below in the docs?

See the https://github.com/openclimatedata/pymagicc/tree/update-docs branch for an example ... rename the file/section to whatever makes sense ...

We should probably have a section "Development" there ... where this info would also go ...

**Comment by znicholls** [comment]
@rgieseke done from my end, pending any changes you want, ready to merge

**Pull request merged**